packages/web-components/src/components/cbp-badge/cbp-badge.scss

```diff
@@ -20,6 +39,6 @@ cbp-badge {
   line-height: var(--cbp-space-6x);
 
   &[color=danger] {
-    --cbp-badge-color-background: var(--cbp-color-danger-base);
+    background-color: var(--cbp-badge-color-background-danger);
```

Comment:
This is going against the principle of setting the CSS property once. Overriding the variable(s) - light and dark - would be the proper way.